pytorch_lightning/metrics/functional/psnr.py

```diff
@@ -31,21 +60,28 @@ def psnr(
     data_range: Optional[float] = None,
     base: float = 10.0,
     reduction: str = 'elementwise_mean',
+    dim: Optional[Union[int, Tuple[int, ...]]] = None,
 ) -> torch.Tensor:
     """
     Computes the peak signal-to-noise ratio
 
     Args:
         preds: estimated signal
         target: groun truth signal
-        data_range: the range of the data. If None, it is determined from the data (max - min)
+        data_range:
+            the range of the data. If `None`, it is determined from the data (max - min). `data_range` must be given
+            when `dim` is not `None`. (default: `None`)
         base: a base of a logarithm to use (default: 10)
         reduction: a method to reduce metric score over labels.
 
             - ``'elementwise_mean'``: takes the mean (default)
             - ``'sum'``: takes the sum
             - ``'none'``: no reduction will be applied
 
+        dim:
+            Use the mean values of squared errors in the given dimension `dim` to calculate PSNR. If `dim` is a list of
+            dimensions, use the mean values over all of them. If `dim` is `None`, use the mean value over all
+            dimensions. (default: `None`)
```

Comment:
Lets reformulate so we use "reduce" instead of "mean"
```suggestion
        dim:
            Dimensions to reduce PSNR scores over, provided as either a integer or list of integers. Default is `None` meaning scores will be reduced across all dimensions.
```

Reply:
It's much clearer! Thank you.